src/main/java/org/cryptomator/integrations/revealpaths/RevealPathsService.java

```diff
@@ -0,0 +1,32 @@
+package org.cryptomator.integrations.revealpaths;
+
+import org.cryptomator.integrations.common.IntegrationService;
+import org.cryptomator.integrations.common.IntegrationsLoader;
+
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.stream.Stream;
+
+public interface RevealPathsService extends IntegrationService {
+
+	/**
+	 * Loads all supported service implementations.
+	 *
+	 * @return Stream of supported RevealPathsService implementations (may be empty)
+	 */
+	static Stream<RevealPathsService> get() {
+		return IntegrationsLoader.loadAll(RevealPathsService.class).filter(RevealPathsService::isSupported);
+	}
+
+	/**
+	 * Opens the parent of the given path in the system default file manager and highlights the resource the path points to.
+	 *
+	 * @param p Path to reveal
+	 * @throws RevealFailedException             If the file manager could not be opened or {@code p} does not have a parent
+	 * @throws java.nio.file.NoSuchFileException If {@code p} does not exist
+	 */
+	//TODO: Throw IllegalArgumenException if p.getParent() == null?
```

Comment:
I'm not sure if this exception handling is too detailed. If we use `open` on macOS, that's how it looks like:

```sh
~ $ open -R foo
The file /Users/<redacted>/foo does not exist.
~ $ echo $?
1
~ $ open -R foo:bar
Unable to reveal file foo:bar.
~ $ echo $?
1
```

And I'm certain that we won't do any string parsing to distinguish between "reveal failed" and "no such file". Maybe just keep `RevealFailedException` and use it for everything? And in the main app, just silently log these failures?

Reply:
The JDK offers a very simple function for this : `Files.exists(Path p)`. No need to parse output.

> And in the main app, just silently log these failures?

This should be discused in the PR of the main app.

Reply:
For documentation: I added the NoSuchFileException, because in Windows the explorer opens anyways, wether the file exists or not.